app/uk/gov/hmrc/gform/core/parsers/SelectionCriteriaParser.scala

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.gform.core.parsers
+
+import uk.gov.hmrc.gform.core.Opt
+import uk.gov.hmrc.gform.sharedmodel.formtemplate.{ CsvColumnName, FormComponentId, FormCtx, SelectionCriteriaValue }
+import uk.gov.hmrc.gform.sharedmodel.formtemplate.SelectionCriteriaValue._
+import uk.gov.hmrc.gform.core.parsers.BasicParsers._
+
+object SelectionCriteriaParser {
+
+  def validate(expression: String): Opt[SelectionCriteriaValue] = validateWithParser(expression, selectionCriteria)
+
+  lazy val selectionCriteria = (
+    ExprParsers.expr ^^ { (_, expr) =>
+      SelectionCriteriaExpr(expr)
+    }
+      |
+        FormatParser.fieldId ~ "." ~ FormatParser.alphabeticOnly ^^ { (_, id, _, column) =>
```

Comment:
We should keep consistency and instead of `FormatParser.fieldId` we should use 
```
FormComponentId.unanchoredIdValidation
``` 
as in all other places.

You just need some implicit magic to make that compile, so don't forget to include:
```
import parseback._
```